examples/flutter_gallery/lib/demo/pesto_demo.dart

```diff
@@ -0,0 +1,683 @@
+              selected: config.showFavorites,
+              onPressed: () { _showFavorites(context); }
+            )
+          ])
+        ),
+        floatingActionButton: new FloatingActionButton(
+          child: new Icon(icon: Icons.edit),
+          onPressed: () { }
+        ),
+        body: new Block(
+          padding: new EdgeInsets.only(top: _kAppBarHeight + statusBarHeight),
+          children: <Widget>[
+            new Padding(
+              padding: const EdgeInsets.all(8.0),
+              child: new MaxTileWidthGrid(
+                maxTileWidth: 500.0,
+                children: recipes.map(
+                  (Recipe r) => _buildCard(context, r)
+                ).toList()
+              )
+            )
+          ]
+        )
+      )
+    );
+  }
+
+  Widget _buildCard(BuildContext context, Recipe recipe) {
+    final TextStyle titleStyle = _textStyle(24.0, FontWeight.w600);
+    final TextStyle authorStyle = _textStyle(12.0, FontWeight.w500).copyWith(color: Colors.black54);
+
+    return new Padding(
+      padding: const EdgeInsets.all(4.0),
+      child: new GestureDetector(
+        onTap: () => _showRecipe(context, recipe),
+        child: new Card(
+          child: new Column(
+            crossAxisAlignment: CrossAxisAlignment.start,
+            children: <Widget>[
+              new AssetImage(
+                name: recipe.imagePath,
+                fit: ImageFit.scaleDown
+              ),
+              new Flexible(child:
+                new Row(
+                  children: <Widget>[
+                    new Padding(
+                      padding: const EdgeInsets.all(16.0),
+                      child: new AssetImage(
+                        width: 48.0,
+                        height: 48.0,
+                        name: recipe.ingredientsImagePath
+                      )
+                    ),
+                    new Column(
+                      crossAxisAlignment: CrossAxisAlignment.start,
+                      mainAxisAlignment: MainAxisAlignment.center,
+                      children: <Widget>[
+                        new Text(recipe.name, style: titleStyle),
+                        new Text(recipe.author, style: authorStyle),
+                      ]
+                    )
+                  ]
+                )
+              )
+            ]
+          )
+        )
+      )
+    );
+  }
+
+  void _showFavorites(BuildContext context) {
+    Navigator.push(context, new MaterialPageRoute<Null>(
+      builder: (BuildContext context) {
+        return new PestoDemo(showFavorites: true);
+      }
+    ));
+  }
+
+  void _showRecipe(BuildContext context, Recipe recipe) {
+    Navigator.push(context, new MaterialPageRoute<Null>(
+      builder: (BuildContext context) {
+        return new Theme(
+          data: _kTheme,
+          child: new _RecipePage(recipe: recipe)
+        );
+      }
+    ));
+  }
+}
+
+class _RecipePage extends StatefulWidget {
```

Comment:
This widget's build function is pretty big. The code might be easier to read if you factored out a widget or two.